Sources/Controllers/Map/OAMapViewController.mm

```diff
@@ -2121,6 +2127,36 @@ - (void) resetViewAngle
     _map3DModeElevationAngle = _mapView.elevationAngle;
 }
 
+- (void)onMapZoomChanged:(id)observable withKey:(id)key andValue:(id)value
+{
+    @synchronized(_rendererSync)
+    {
+        [self updateMapLocaleLanguage];
+    }
+}
+
+- (void)updateMapLocaleLanguage
+{
+    NSString *langId = [self getMapPreferredLocale:_mapView.zoomLevel];
+    if (_mapPresentationEnvironment != nullptr && ![langId isEqualToString:_mapPresentationEnvironment->getLocaleLanguageId().toNSString()])
```

Comment:
It is more optimal to first check _mapPresentationEnvironment != nullptr, and then call NSString *langId = ...